bin/echo.c

```diff
@@ -177,31 +160,33 @@ int negotiate(struct s2n_connection *conn, int fd)
     printf("Curve: %s\n", s2n_connection_get_curve(conn));
     printf("KEM: %s\n", s2n_connection_get_kem_name(conn));
     printf("KEM Group: %s\n", s2n_connection_get_kem_group_name(conn));
+    printf("Cipher negotiated: %s\n", s2n_connection_get_cipher(conn));
+
+    if (session_resumed) {
+        printf("Resumed session\n");
+    }
 
     uint32_t length;
     const uint8_t *status = s2n_connection_get_ocsp_response(conn, &length);
     if (status && length > 0) {
         fprintf(stderr, "OCSP response received, length %u\n", length);
     }
+    return 0;
+}
 
-    printf("Cipher negotiated: %s\n", s2n_connection_get_cipher(conn));
-
-    bool session_resumed = s2n_connection_is_session_resumed(conn);
-    if (session_resumed) {
-        printf("Resumed session\n");
-    }
-
+void psk_early_data(struct s2n_connection *conn,  bool session_resumed)
```

Comment:
Does the return type need to be an int? I'm confused how the GUARDs in this function are working.

Reply:
Speaking of `GUARD`: for `s2nd`/`s2nc` I think we've tried to not use any internal APIs so it's closer to what customers will be doing and can use it as an example and starting point. I'm not sure how well we've stuck to that, though. @lrstewart may need to speak to that more.

If we are using internal APIs, then we should return `S2N_RESULT` rather than `int`, as it's a lot more strict and prevents this exact kind of issue (returning an `int` in a `void`)

Reply:
Ah, reading it more it only calls `GUARD_EXIT`, which completely exits the program:
https://github.com/aws/s2n-tls/blob/5d8b6be7b5b739cbf646e5cd2d42f050823afbb3/bin/common.h#L29-L35